packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -0,0 +1,111 @@
+    }
+
+    async renderExtension(container: string, region: ExtensionRegion): Promise<void> {
+        const extension = this._getExtensionByRegion(region);
+
+        if (!extension) {
+            return;
+        }
+
+        await this._checkoutService.renderExtension(container, region);
+
+        if (!this._handlerRemovers[extension.id]) {
+            this._handlerRemovers[extension.id] = [];
+        }
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.ReloadCheckout,
+                () => {
+                    void this._checkoutService.loadCheckout(
+                        this._checkoutService.getState().data.getCheckout()?.id,
+                    );
+                },
+            ),
+        );
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.SetIframeStyle,
+                (data) => {
+                    if (data.type === ExtensionCommandType.SetIframeStyle) {
+                        const { style } = data.payload;
+                        const extensionContainer = document.querySelector(
+                            `div[data-extension-id="${extension.id}"]`,
+                        );
+                        const iframe = extensionContainer?.querySelector('iframe');
+
+                        if (iframe) {
+                            Object.assign(iframe.style, style);
+                        }
+                    }
+                },
+            ),
+        );
+
+        this._handlerRemovers[extension.id].push(
+            this._checkoutService.handleExtensionCommand(
+                extension.id,
+                ExtensionCommandType.ShowLoadingIndicator,
+                (data) => {
+                    if (data.type === ExtensionCommandType.ShowLoadingIndicator) {
+                        const { show } = data.payload;
+
+                        this._setIsShowLoadingIndicator(show);
+                    }
+                },
+            ),
+        );
+    }
+
+    removeListeners(region: ExtensionRegion): void {
+        const extension = this._getExtensionByRegion(region);
+
+        if (!extension) {
+            return;
+        }
+
+        const removers = this._handlerRemovers[extension.id];
+
+        if (!removers) {
+            return;
+        }
+
+        for (const remover of removers) {
+            remover();
+        }
+
+        delete this._handlerRemovers[extension.id];
+    }
+
+    isRegionInUse(region: ExtensionRegion): boolean {
+        const extension = this._getExtensionByRegion(region);
+
+        return Boolean(extension);
+    }
+
+    private _getExtensionByRegion(region: ExtensionRegion): Extension | undefined {
+        const extensions = this._checkoutService.getState().data.getExtensions();
```

Comment:
Any reason we are not using `getExtensionByRegion` selector?

Reply:
`getExtensionByRegion` is a pirvate selector in SDK 😅, only `getExtensions` is public available.

Reply:
Hmm I think might be a good idea to make it available outside SDK as the clients as we can see here need to duplicate that logic.

Reply:
I want to avoid SDK depency in this PR.
Will do it right after this PR, here is a [ticket](https://bigcommercecloud.atlassian.net/browse/CHECKOUT-7634.) to track it.

Reply:
Sounds perfect, we can do it as a fast follow 😄